src/Microsoft.DotNet.Interactive/Server/KernelCommandEnvelope.cs

```diff
@@ -25,6 +25,10 @@ static KernelCommandEnvelope()
             ResetToDefaults();
         }
 
+        internal static bool IsCommandTypeRegistered(Type commaType)
+        {
+            return _commandTypesByCommandTypeName.ContainsKey(commaType.Name);
```

Comment:
I didn't realize this was static. This should really all be kernel-level state.

Reply:
that is a registration for serialization is static with the envelope not the kernel, do we need to track these kind of registration in more than one place?

Reply:
Or shift them to the kernels maybe? Just trying to reduce global state.

Reply:
The envelope serializers will have to know that state

Reply:
removed that